### DISCUSSION TITLE
[Feature] Transcode policy for images (e.g. high res for heic only)

### DISCUSSION THREAD

**Discussion by pyorot** [immich-app/immich#10542] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Last transcoding request I promise 😇. Unfortunately, things are moving in the direction where universal support of image formats cannot be expected, most notably with heic only working in safari. As such, I think it will be useful to refactor server image settings to be more like video (transcoding) settings, but still with thumbnails and previews.

A basic use-case: we want low-quality previews for fast browsing. We also want high-quality originals for zooming in. I personally have "prefer originals" enabled and so keep low-quality on the server but usually preview the original assets. However, this breaks down for heic format images. These can't be opened in original format in chrome (etc. except safari), so you are obliged to either (1) use the low-quality previews for zoom or (2) transcode all your heic AND jpeg assets into high-quality previews (so collateral damage on fast loads and low storage requirement for previews of jpeg assets). Here, a quick fix is transcode jpeg assets into low-quality jpeg previews and heic into high quality jpeg previews.

Down the line: split previews from transcoded originals (so have 5 tiers: blurred thumbnail, thumbnail, preview, transcoded full, original asset), and so have heic assets cover all 5 tiers in principle, while jpeg assets can cover all but "transcoded full".

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
We've briefly discussed wanting to use pyramid images so you can load a lower resolution version initially and scale to higher resolution tiles as you zoom. I imagine this would allow more flexibility in the resolution of the initial preview.

**Comment by Eskuero** [comment]
I already do manually compress my pictures to avif before uploading to immich but it would be nice to have a policy for my other users in the same style as videos
